CustomerReviewsModule.Data/Services/CustomerReviewService.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Linq;
+using CustomerReviewsModule.Core.Models;
+using CustomerReviewsModule.Core.Services;
+using CustomerReviewsModule.Data.Models;
+using CustomerReviewsModule.Data.Repositories;
+using VirtoCommerce.Platform.Core.Common;
+using VirtoCommerce.Platform.Data.Infrastructure;
+
+namespace CustomerReviewsModule.Data.Services
+{
+    public class CustomerReviewService : ServiceBase, ICustomerReviewService
+    {
+        private readonly Func<ICustomerReviewRepository> _repositoryFactory;
+
+        public CustomerReviewService(Func<ICustomerReviewRepository> repositoryFactory)
+        {
+            _repositoryFactory = repositoryFactory;
+        }
+
+        public CustomerReview[] GetByIds(string[] ids)
+        {
+            using (var repository = _repositoryFactory())
+            {
+                return repository
+                    .GetByIds(ids)
+                    .Select(x => x.ToModel(AbstractTypeFactory<CustomerReview>.TryCreateInstance()))
+                    .ToArray();
+            }
+        }
+
+        public void SaveCustomerReviews(CustomerReview[] items)
+        {
+            if (items == null)
+                throw new ArgumentNullException(nameof(items));
+
+            var pkMap = new PrimaryKeyResolvingMap();
+            using (var repository = _repositoryFactory())
+            {
+                using (var changeTracker = GetChangeTracker(repository))
+                {
+                    var alreadyExistEntities = repository
+                        .GetByIds(items.Where(m => !m.IsTransient())
+                        .Select(x => x.Id)
+                        .ToArray());
+                    foreach (var derivativeContract in items)
+                    {
+                        var sourceEntity = AbstractTypeFactory<CustomerReviewEntity>.TryCreateInstance().FromModel(derivativeContract, pkMap);
+                        var targetEntity = alreadyExistEntities.FirstOrDefault(x => x.Id == sourceEntity.Id);
+                        if (targetEntity == null)
+                        {
+                            repository.Add(sourceEntity);
+                            continue;
```

Comment:
It took a while to understand this logic here.
IMHO, we should avoid using "continue;" and add this rule to our styleguide.

Reply:
Agree, not such representative example!) But basically I try not to use full "if..else" syntax, because it increase nesting very often. Instead, I just invert "if" conditions if possible, and thereby remove "else" part.